packages/web3-core/src/web3_context.ts

```diff
@@ -98,9 +104,23 @@ export class Web3Context<
 				this.requestManager,
 				registeredSubscriptions,
 			);
+		} else {
+			this._subscriptionManager = new Web3SubscriptionManager(
+				this.requestManager,
+				Web3Context.getDefaultRegisteredSubscriptions(),
+			);
 		}
 	}
 
+	private static getDefaultRegisteredSubscriptions(): any {
+		return {
+			logs: LogsSubscription,
+			newPendingTransactions: NewPendingTransactionsSubscription,
+			newHeads: NewHeadsSubscription,
+			syncing: SyncingSubscription,
+		};
+	}
```

Comment:
This could be false positive, as the user can pass `registeredSubscriptions` to the context, even user can pass `subscriptionManager`. So the values returned by this function will be false positive.